### PR TITLE
Panic! in the Build System (Use new library for Travis runner)

### PR DESCRIPTION
**Pull request by alexwlchan**
This PR migrates our Travis test runner to use the new `travistooling` library. Bonus is more testing, more reliability, and we can start acting on the Travis-ECR notifications to improve build times by patching the new library.

This is largely driven by thinking around https://github.com/HypothesisWorks/hypothesis/issues/1224 – I’d really like to have us using travistooling and getting it working properly _before_ I try to use it in another project!

### CODE REVIEW

**Comment by alexwlchan** [comment]
Oh, and I ditched the `.travis` directory, because dotfiles being hidden on macOS was just making that mildly tedious.

**Comment by alexwlchan** [comment]
Niiiiiice.

screen shot 2018-04-14 at 17 59 28

**Comment by alexwlchan** [comment]
In general, changes to the Travis build scripts (travistooling, .travis, .travis.yml) probably shouldn’t trigger a full rebuild – but I’m going to save that for a separate patch, as I want to check we can still do a full build with this setup.

File: run_travis_task.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                  | Relevant changes | No relevant changes
+   ---------------+------------------+---------------------
+    pull request  | Run tests        | Don't run tests
+                  | Don't deploy     | Don't deploy
+   ---------------+------------------+---------------------
+    master        | Run tests        | Run tests
+                  | Deploy to AWS    | Don't deploy
+
+We always run tests on master so we get consistent build results, that's
+less important on master where results are transient.
```
Comment (alicefuzier) [inline run_travis_task.py]: I think you meant less important on pull requests?

File: run_travis_task.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    publish_task = task.replace('-build', '-publish')
+    publish_task = task.replace('-test', '-publish')
+
+    if _should_publish(task=task, travis_event_type=travis_event_type):
+        print("*** We're going to run the publish task")
+        make(publish_task)
+    else:
+        print("*** We don't need to run the publish task")
+        make(publish_task, '--dry-run')
```
Comment (alicefuzier) [inline run_travis_task.py]: Why do we do a dry run here rather than just return 0?

File: run_travis_task.py (hunk 1 above)
Comment (alexwlchan) [reply run_travis_task.py]: Oops. Yes, will fix.

File: run_travis_task.py (hunk 2 above)
Comment (alexwlchan) [reply run_travis_task.py]: Doing dry-run ensures the task actually exists, so we don’t get errors on master if we’ve defined a test task but no publish task. Will add a comment.

**Pull request merged**